### DISCUSSION TITLE
Is there any plan to simplify redundancy? I have to add the same translation to multiple places

### DISCUSSION THREAD

**Discussion by d4h0y0sr** [DivinumOfficium/divinum-officium#3835] [Developers corner]
Hi, while adding translations i found out i need to add "almost" same information to multiple places. Lets take the psalms, the psalms are in a folder but they have a numeration, then i can not used for the mass files because i dont want to include that information. Or the evangelium, they appear both in mass and horas, but they are part of sections that include homily or other elements that made them difficult to reference. I wish there were an easier design to not repeat same information in multiple places.

**Comment by FAJ-Munich** [comment]
Yes, there are plans to reduce redundancy as much as possible. Whenever you see a possibility to use an @ reference to some other file where the text is "more proper" to, please make use of it, preferably in the Latin. Then you can delete all the redundant hashes in the vernaculars.

**Comment by FAJ-Munich** [comment]
@mbab @APMarcello3 

As a means of reducing redundancies and making the database text files more manageable across versions, how about merging the Missa hashes (all the database files in Sancti, Tempora, and Commune) into the files used for horas? 
Since the Tabulae is already shared across missa and horas, this hopefully helps to not forget one or the other when correcting errors, e.g., of commemorations and such.

**Comment by APMarcello3** [comment]
@FAJ-Munich I am not opposed to this; however, I would first like to see addressed #1086 and #838. Issues of the calendar and accuracy do have highest priority.

**Reply by FAJ-Munich** [reply]
Regarding #838, what has been decided for the verse numbering in the Latin-Bea version?

**Reply by APMarcello3** [reply]
@FAJ-Munich The Pius XII Psalter verse numbering can be found in the Colunga-Turrado Biblia Vulgata, published by BAC: https://archive.org/details/BIBLIASACRAIUXTAVULGATAMCLEMENTINAMCOLUNGAETTURRADO

**Comment by fiapps** [comment]
When eliminating redundancy, it would be good to give thought to which redundant version to preserve instead of simply keeping whichever one happens to be in the common, for example. Over the years I've often seen cases where accents are added to a Latin proper text, and then a couple of years later the proper text is deleted because it's the same as the common. But the common had never been given accents, so the elimination of redundancy caused the accents to disappear from the text users see. Another case is the translations. The original English translations were in an old-fashioned English that over the years Fr. Marcello modernized. But when he modernized a proper text, that work was later lost when the proper text was deleted because it was another translation of the common (which hadn't been modernized). The sort of thing happened especially with commons that are only used for a small number of feasts.

So when you eliminate redundancy, it would be good to consider whether one of the redundant texts is of superior quality.

**Reply by FAJ-Munich** [reply]
Thanks for pointing that out. Important reminder indeed.

**Comment by APMarcello3** [comment]
@fiapps @FAJ-Munich The question of which translations are used is ultimately my decision. Some archaisms I have chosen to remove in favor of more intelligible but equally open-source translations. I realize these may not always be popular decisions. Please consult first with me on these matters. Indeed, all major projects (particularly simplification) should receive my approval before work is undertaken. Thank you!

**Reply by fiapps** [reply]
@APMarcello3 of course the decision is yours. I was pointing out that your work has at times been reversed when commons you didn't modify were de facto selected in preference to proper texts you did modify. Thus the need for care and perhaps your personal attention when decisions are made about which variant of a text to retain.

**Reply by FAJ-Munich** [reply]
I have created a little perl script to create a "sitemap" of DO database into a single CSV file. 
As of today, we have 29569 unique hashes in the horas and missa folders for Commune, Sancti, and Tempora worth 48 MB of data. Sorting the data in Excel, I hope it becomes more easily to spot deviating translations and manage editing Hashes across all vernaculars. Of course, it becomes useful only when I managed to program a way to re-distribute the edited file back into the database logic. Let's see! Since the max size for attachments is 25 MB I cannot share it here as of now.

**Reply by APMarcello3** [reply]
@FAJ-Munich I would be interested if you could kindly share with me. Please contact me at canon DOT missae AT gmail DOT com.

**Reply by FAJ-Munich** [reply]
While revising the Commons, I've come across diverging translations in **Spanish, French, and Hungarian** in _Commune Virginum C6_ for the same Latin texts in _Versicles at Matins_ vs. _R.br. and V. at the Minor Hours_. Guidance appreciated!

![CommuneC6](https://github.com/user-attachments/assets/4d61c51d-1986-49cf-af4b-9f51efb3b127)

**Reply by d4h0y0sr** [reply]
> While revising the Commons, I've come across diverging translations in **Spanish, French, and Hungarian** in _Commune Virginum C6_ for the same Latin texts in _Versicles at Matins_ vs. _R.br. and V. at the Minor Hours_. Guidance appreciated!
> 
> ![CommuneC6](https://private-user-images.githubusercontent.com/60485740/365796634-4d61c51d-1986-49cf-af4b-9f51efb3b127.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5MTQ1MTUsIm5iZiI6MTcyNTkxNDIxNSwicGF0aCI6Ii82MDQ4NTc0MC8zNjU3OTY2MzQtNGQ2MWM1MWQtMTk4Ni00OWNmLWFmNGItOWY1MWVmYjNiMTI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDIwMzY1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyODU0ZjAyYjY5OGU1MjNhZTQwNmNiYzA2OGY0ZjAxOTM1NTEwZDE5NWU2NjQ4ZDhkMmUyYjY2NWM1NWI1MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vBxhsnYPY1dfRdHKWSQicqi4M1u9JsfscEj5pzfHe7g)

I could help with spanish, could you elaborate on the issue?

**Reply by FAJ-Munich** [reply]
> > While revising the Commons, I've come across diverging translations in **Spanish, French, and Hungarian** in _Commune Virginum C6_ for the same Latin texts in _Versicles at Matins_ vs. _R.br. and V. at the Minor Hours_. Guidance appreciated!

> 
> I could help with spanish, could you elaborate on the issue?

In the Commons, the Versicles at the 3 Nocturns are usually identical to the text used for the short Responsories and Versicles at the minor hours. When you compare the bold translations in the screenshot attached above (composed from C6.txt files) you can see that the Spanish translations in Matins are different to the ones used in the Minor Hours. For consistency, the question is which one to keep and which one to purge.

**Reply by FAJ-Munich** [reply]
@d4h0y0sr  in Spanish, additionally, there are two translations of the last two stanzas of the Hynus Matutinum in C6 for Martyr and non-Martyr Virgins:

One reads:
```
Por sus plegarias, dígnate, Dios sumo, 
Nuestras ofensas perdonar propicio; 
Libres de vicio para que a Ti alcemos 
Puras las voces. 
_ 
Honra perpetua al Padre y Unigénito, 
Y al procedente Espíritu divino; 
Dios uno y trino: honor, imperio y gloria, 
Siglos eternos. 
Amén.
```

The other:
```
Dígnate, oh Dios de bondad, perdonarnos,  
por los méritos de esta Santa, las penas  
merecidas por nuestros pecados, para que te cantemos  
santos himnos con corazón puro. 
_
 Alabanza sea dada a ti, oh Padre,  
y a tu Hijo Unigénito 
 juntamente con el Espíritu Consolador,  
por los siglos de los siglos.  
Amén.
```

The Latin is identical `Hujus orátu, ...`
